services/question/README.md

```diff
@@ -0,0 +1,428 @@
+      "difficulty": "Easy",
+      "title": "Reverse a String",
+      "topics": [
+        "Strings",
+        "Algorithms"
+      ]
+    },
+    {
+      "_id": "66ea6985cd34132719540c23",
+      "id": 2,
+      "description": "Implement a function to detect if a linked list contains a cycle.",
+      "difficulty": "Easy",
+      "title": "Linked List Cycle Detection",
+      "topics": [
+        "Data Structures",
+        "Algorithms"
+      ]
+    }
+  ]
+}
+```
+
+---
+
+## Get Question by ID
+
+This endpoint allows the retrieval of the question by using the question ID.
+
+- **HTTP Method**: `GET`
+- **Endpoint**: `/questions/{questionId}`
+
+### Parameters:
+
+- `questionId` (Required) - The ID of the question to retrieve.
+
+### Responses:
+
+| Response Code               | Explanation                                                     |
+|-----------------------------|-----------------------------------------------------------------|
+| 200 (OK)                    | Success, question corresponding to the `questionID` is returned |
+| 404 (Not Found)             | Question with the specified `questionID` not found              |
+| 500 (Internal Server Error) | Unexpected error in the database or server                      |
+
+### Command Line Example:
+
+```
+Retrieve Question by ID:
+curl -X GET http://localhost:8081/questions/1
+```
+
+### Example of Response Body for Success:
+
+```json
+{
+  "status": "Success",
+  "message": "Question with ID retrieved successfully",
+  "data": {
+    "_id": "66ea6985cd34132719540c22",
+    "id": 1,
+    "description": "Write a function that reverses a string. The input string is given as an array of characters s.\n\nYou must do this by modifying the input array in-place with O(1) extra memory.\n\n\nExample 1:\n\nInput: s = [\"h\",\"e\",\"l\",\"l\",\"o\"]\nOutput: [\"o\",\"l\",\"l\",\"e\",\"h\"]\n\nExample 2:\nInput: s = [\"H\",\"a\",\"n\",\"n\",\"a\",\"h\"]\nOutput: [\"h\",\"a\",\"n\",\"n\",\"a\",\"H\"]\n\nConstraints:\n1 \u003C= s.length \u003C= 105 s[i] is a printable ascii character.",
+    "difficulty": "Easy",
+    "title": "Reverse a String",
+    "topics": [
+      "Strings",
+      "Algorithms"
+    ]
+  }
+}
+```
+
+---
+
+## Get Question by Parameters (Random)
+
+This endpoint allows the retrieval of random questions that matches the parameters provided.
+
+- **HTTP Method**: `GET`
+- **Endpoint**: `/questions/search`
+
+### Parameters:
+
+- `limit` (Optional) - The number of questions to be returned. If not provided, default limit is 1.
```

Comment:
Sorry I thought we were excluding the `limit` field from this endpoint?

Reply:
Hm, I included the limit field because it allows us to control the number of questions returned after filtering, which I think is useful for admins. It also enables us to randomly select questions within that limit. 

Then for users, the default is set to one random question, so I don't think they will be affected by this.